src/vs/editor/contrib/smartSelect/browser/smartSelect.ts

```diff
@@ -311,10 +312,17 @@ CommandsRegistry.registerCommand('_executeSelectionRangeProvider', async functio
 	assertType(URI.isUri(resource));
 
 	const registry = accessor.get(ILanguageFeaturesService).selectionRangeProvider;
+	const configurationService = accessor.get(IConfigurationService);
 	const reference = await accessor.get(ITextModelService).createModelReference(resource);
 
+	const options: SelectionRangesOptions = configurationService.getValue<SelectionRangesOptions | undefined>('editor.smartSelect', { resource }) ?? {
```

Comment:
You also need to pass `overrideIdentifier` which would be the language id (or use the `ITextResourceConfigurationService` which does this for you)

Reply:
Using `ITextResourceConfigurationService` meant that I needed to add `TestTextResourceConfigurationService` to the tests, and it seems that `TestTextResourceConfigurationService#updateValue` is currently a no-op:

https://github.com/microsoft/vscode/blob/cf3d478c3184c8065a1cc25a81c95a54672a7cd0/src/vs/platform/configuration/test/common/testConfigurationService.ts#L45-L46

This means I can't set the config from my test. If I understand correctly, we would need to fix that in order to test this. Do I understand correctly?

I also attempted to use `IConfigurationService`, passing along `overrideIdentifier`, however it's not clear how I would infer the language. Should we make this an extra argument to the command? Or should we try to infer it from the resource similar to `TextResourceConfigurationService#getLanguage`?

Reply:
In the meantime, I managed to get this working: 89c2c1e76e26961c72dc82af2dbcbf2225208472

Essentially we're reading language overrides but we're only testing user config. Ideally the test would also check that it respects language overrides.